docs/source/deployment/CORS_Headers.rst

```diff
@@ -0,0 +1,173 @@
+========================================
+Add CORS headers to server configuration
+========================================
+
+.. contents:: Table of Contents
+    :depth: 3
+
+Overview
+========
+
+If the MOTECH-UI is hosted separately from MOTECH-CORE, we have to share resources between different domains.
+XMLHttpRequest requests have traditionally been limited to accessing the same domain as the parent web page (as per the same-origin security policy).
+Cross-site requests are forbidden by default because of their ability to perform advanced requests that introduce many cross-site scripting security issues.
+Cross-Origin Resource Sharing (CORS) can define a way in which MOTECH-UI and MOTECH-CORE interact to determine safely whether or not to allow the cross-origin request.
+
+It gives possibilities to specify which domains will have access to resources. This documentation page explains how MOTECH-CORE can configure its headers to support CORS.
+
+Cross-site requests
+===================
+
+If we want to share resources, the MOTECH-CORE(server) must enable CORS.
+However sending a cross-site request does not require setting any cross-origing sharing request headers programmatically.
+CORS headers are set for you when making invocations to server. Here you can find more information about `the HTTP request headers <https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#The_HTTP_request_headers>`_.
+
+Enable CORS in MOTECH-CORE
+==========================
+
+By default when a cross-site request is sent to MOTECH-CORE, the following error occurs :
+
+.. code-block::
+
+	Origin http://example.domain.com is not allowed by Access-Control-Allow-Origin.
+
+To enable CORS we have to change configuration in used web server. We can add the following CORS headers :
+
+Access-Control-Allow-Origin
+----------------------------
+
+In this header we can specify which domains have access to resources. We can allow access from any origin using "*". But if you’d like finer control over who can access your data, use an actual value in the header. Examples:
+
+.. code-block::
+
+   Access-Control-Allow-Origin: "*" (allow access from any origin)
+   Access-Control-Allow-Origin: "http://motech-ui.example" (allow access from only "http://motech-ui.example" origin)
+   Access-Control-Allow-Origin: "http://motech-ui.example | http://other.domain" (allow access from two mentioned origins)
+
+Access-Control-Allow-Methods
```

Comment:
Do we have to add this if we add the allow-origin header? I imagine in most use cases I would want to allow all methods (we don't use CONNECT, TRACK, TRACE and OPTIONS but I doubt anyone will really care)